app/containers/LoginOrRegister.jsx

```diff
@@ -17,28 +17,27 @@ class LoginOrRegister extends Component {
   constructor(props) {
     super(props);
     this.toggleMode = this.toggleMode.bind(this);
-    this.onLoginSubmit = this.onLoginSubmit.bind(this);
-    this.onRegisterSubmit = this.onRegisterSubmit.bind(this);
+    this.handleOnSubmit = this.handleOnSubmit.bind(this);
   }
 
-  onLoginSubmit() {
-    const { dispatch } = this.props;
-    const email = ReactDOM.findDOMNode(this.refs.email).value;
-    const password = ReactDOM.findDOMNode(this.refs.password).value;
-    dispatch(manualLogin({
-      email,
-      password
-    }));
-  }
+  handleOnSubmit(event) {
+    event.preventDefault();
 
-  onRegisterSubmit() {
-    const { dispatch } = this.props;
+    const { dispatch, user: { isLogin } } = this.props;
     const email = ReactDOM.findDOMNode(this.refs.email).value;
     const password = ReactDOM.findDOMNode(this.refs.password).value;
-    dispatch(signUp({
-      email,
-      password
-    }));
+
+    if (isLogin) {
+      dispatch(manualLogin({
+        email,
+        password
+      }));
+    } else {
+      dispatch(signUp({
```

Comment:
I'm keeping the logic the same as the boilerplate. The isLogin property on the state determines whether or not the user is trying to login or register. When it is true, the user is trying to login. When it is false, the user is trying to register. Take a look at the old renderButton method on this container, the logic will look similar. The action creator toggleLoginMode toggles between the two mode, and there is only two mode in this case.